Source/Core/Core/Boot/DolReader.cpp

```diff
@@ -94,6 +96,17 @@ bool DolReader::Initialize(const std::vector<u8>& buffer)
     }
   }
 
+  // Check if this dol contains an ancast image
+  // The ancast image will always be in the first data section
+  m_is_ancast = false;
+  if (m_data_sections[0].size() > sizeof(EspressoAncastHeader) && 
+      m_dolheader.dataAddress[0] == ESPRESSO_ANCAST_LOCATION_VIRT)
+  {
+    // Check for the ancast magic
+    if (*((u32*)m_data_sections[0].data()) == Common::swap32(ANCAST_MAGIC))
```

Comment:
Casting to u32* and dereferencing is risky due to aliasing and alignment rules. I think in this specific case it might not be UB, but we have a cleaner way of doing this kind of thing:

```suggestion
    if (Common::swap32(m_data_sections[0].data()) == ANCAST_MAGIC)
```

Reply:
> pedantic, but it should be some big-endian reader instead of always swapping

Big-endian support seems out-of-scope for this PR, considering that the rest of the DolReader also swaps the entire dol header :P